kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoWriter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.db.mongo;
+
+import dev.responsive.kafka.internal.db.RemoteTable;
+import dev.responsive.kafka.internal.db.RemoteWriter;
+import dev.responsive.kafka.internal.stores.RemoteWriteResult;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+public class MongoWriter<K> implements RemoteWriter<K> {
+
+  private final RemoteTable<K> table;
+  private final int partition;
+
+  public MongoWriter(final RemoteTable<K> table, final int partition) {
+    this.table = table;
+    this.partition = partition;
+  }
+
+  @Override
+  public void insert(final K key, final byte[] value, final long epochMillis) {
+    table.insert(partition, key, value, epochMillis);
+  }
+
+  @Override
+  public void delete(final K key) {
+    table.delete(partition, key);
+  }
+
+  @Override
+  public CompletionStage<RemoteWriteResult> flush() {
+    return CompletableFuture.completedFuture(RemoteWriteResult.success(partition));
```

Comment:
We may want to either have two different APIs -- one for sync one for async, or just use the Async mongo driver here. We'll need to do some refactoring of the API to make sure that it doesn't have anything C* specific.

Reply:
Just making sure I understand: the C* client is currently/already async, so the choice is between introducing a synchronous API for the mongo client vs plugging in the async mongo client -- is that correct?

If so, two followup questions:
1. why not just plug in the async mongo driver right away?
2. it looks like we're just automatically plugging the results into a completed future/stage in both clients right now, so it's not truly async at the moment anyways. Do we want to make it be truly async at some point in the near future? Like I guess we could do something pretty much exactly the same as the recent changelog truncation optimization -- we don't actually need to know if it failed/was offset fenced during the Kafka commit, since either way we only flush after the txn/offsets are committed to kafka anyways. So we could just kick off a new flush after a commit in the same way that we kick off a new DeleteRecords request, and throw a fenced exception if it turns out the flush did indeed fail

I guess I'm getting ahed of things here, obviously this doesn't need to happen as part of this change. Just trying to envision the future and why we would/would not want the async driver

Reply:
1. Async is generally more complicated and I wanted to optimize for simplicity for the first go around. Also curious about non-async performance since if it's good enough we can just batch the entire thing in the background
2. I don't think that's true for the `LwtWriter` (it is true for the `FactWriter`), which does return a future that isn't complete. I do think we need to know at least before the _next_ flush whether or not it succeeded, so making that optimization is a bit more difficult. We wouldn't want to flush the next batch if the previous batch failed to flush or we'd corrupt the remote store.